flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosResourceManager.java

```diff
@@ -312,6 +312,48 @@ private void recoverWorkers() throws Exception {
 	}
 
 	@Override
+	public void shutDown() throws Exception {
+		// shut down all components
+		if (taskMonitor != null) {
+			try {
+				actorSystem.stop(taskMonitor);
+			} catch (Throwable tt) {
+				LOG.error("Failed to stop taskMonitor", tt);
```

Comment:
We should rather add the caught exceptions as suppressed exceptions to a global exception. See `shutDownApplication` for details.